packages/mobile/src/exchange/ExchangeHomeScreen.tsx

```diff
@@ -1,126 +1,231 @@
-  exchangeRate: BigNumber
-  goldBalance: string | null
-  dollarBalance: string | null
-}
+type Props = StackScreenProps<StackParamList, Screens.ExchangeHomeScreen>
 
-interface DispatchProps {
-  fetchExchangeRate: typeof fetchExchangeRate
-}
+function ExchangeHomeScreen({ navigation }: Props) {
+  const scrollPosition = useRef(new Animated.Value(0)).current
+  const { t } = useTranslation(Namespaces.exchangeFlow9)
+  const dollarBalance = useSelector((state) => state.stableToken.balance)
+  const goldBalance = useSelector((state) => state.goldToken.balance)
+  const dispatch = useDispatch()
 
-type OwnProps = StackScreenProps<StackParamList, Screens.ExchangeHomeScreen>
+  // TODO: revert this back to `useLocalCurrencyCode()` when we have history data for cGDL to Local Currency.
+  const localCurrencyCode = null
+  const localExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const currentExchangeRate = useExchangeRate()
+  const goldToDollars = useCallback((amount) => goldToDollarAmount(amount, currentExchangeRate), [
+    currentExchangeRate,
+  ])
+  const dollarsToLocal = useCallback(
+    (amount) => convertDollarsToLocalAmount(amount, localCurrencyCode ? localExchangeRate : 1),
+    [localExchangeRate]
+  )
+  const displayLocalCurrency = useCallback(
+    (amount: BigNumber.Value) =>
+      getLocalCurrencyDisplayValue(amount, localCurrencyCode || LocalCurrencyCode.USD, true),
+    [localCurrencyCode]
+  )
 
-type Props = StateProps & DispatchProps & WithTranslation & OwnProps
+  const currentGoldRateInLocalCurrency = dollarsToLocal(goldToDollars(1))
+  let rateChangeInPercentage, rateWentUp
+  const exchangeHistory = useSelector(exchangeHistorySelector)
+  if (exchangeHistory.aggregatedExchangeRates.length) {
+    const oldestGoldRateInLocalCurrency = dollarsToLocal(
+      exchangeHistory.aggregatedExchangeRates[0].exchangeRate
+    )
+    if (oldestGoldRateInLocalCurrency) {
+      const rateChange = currentGoldRateInLocalCurrency?.minus(oldestGoldRateInLocalCurrency)
+      rateChangeInPercentage = currentGoldRateInLocalCurrency
+        ?.div(oldestGoldRateInLocalCurrency)
+        .minus(1)
+        .multipliedBy(100)
+      rateWentUp = rateChange?.gt(0)
+    }
+  }
 
-const mapStateToProps = (state: RootState): StateProps => ({
-  exchangeRate: getRateForMakerToken(state.exchange.exchangeRatePair, CURRENCY_ENUM.DOLLAR),
-  goldBalance: state.goldToken.balance,
-  dollarBalance: state.stableToken.balance,
-})
+  const hasGold = new BigNumber(goldBalance || 0).isGreaterThan(0)
 
-export class ExchangeHomeScreen extends React.Component<Props> {
-  componentDidMount() {
-    this.props.fetchExchangeRate()
-  }
+  useEffect(() => {
+    dispatch(fetchExchangeRate())
+  }, [])
+
+  const headerOpacity = useMemo(
+    () => ({
+      opacity: scrollPosition.interpolate({
+        inputRange: [0, 100],
+        outputRange: [0, 1],
+        extrapolate: Animated.Extrapolate.CLAMP,
+      }),
+    }),
+    []
```

Comment:
shouldn't scrollPosition be in this array?

Reply:
As it states here https://reactjs.org/docs/hooks-reference.html#useref, `scrollPosition` will persist for the full lifetime of the component, so it is unnecessary to add it to deps array.